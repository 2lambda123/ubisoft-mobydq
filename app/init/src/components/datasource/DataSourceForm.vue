<template>
  <div>
    <h1 class="mt-5">Edit Data Source</h1>

    <form>
      <div class="form-row">
        <div class="col-md-4">
          <!-- Data Source Form -->
          <div class="form-group required">
            <label for="dataSourceName" class="col-form-label">
              Name:
            </label>
            <input
              class="form-control col-sm"
              id="dataSourceName"
              type="text"
              required="required"
              placeholder="Type data source name"
              v-model="dataSource.name"
            />
          </div>
          <div class="form-group required">
            <data-source-select-data-source-type
              v-model="dataSource.dataSourceTypeId"
              v-on:changeDataSourceType="getDataSourceType">
            </data-source-select-data-source-type>
          </div>
          <div class="form-group required">
            <label for="dataSourceConnectionString" class="col-form-label">
              Connection String:
            </label>
            <textarea
              class="form-control col-sm"
              id="dataSourceConnectionString"
              required="true"
              placeholder="Type data source connection string"
              rows="3"
              v-model="dataSource.connectionString"
            />
          </div>
          <div class="form-group">
            <label for="dataSourceLogin" class="col-form-label">
              Login:
            </label>
            <input
              class="form-control col-sm"
              id="dataSourceLogin"
              type="text"
              placeholder="Type data source login"
              v-model="dataSource.login"
            />
          </div>
          <div v-if="showPasswordField" class="form-group">
            <label for="dataSourcePassword" class="col-form-label">
              Password:
            </label>
            <input
              class="form-control col-sm"
              id="dataSourcePassword"
              type="password"
              placeholder="Type data source password"
              v-model="dataSource.password" />
          </div>
          <div class="form-group">
              Connectivity Status:
              <span class="badge badge-pill" v-bind:class="{ 'badge-success': connectivityStatusOK, 'badge-danger': connectivityStatusKO }" >
                  {{ dataSource.connectivityStatus }}
              </span>
              <span v-show="spinner" class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
          </div>

          <!-- Meta-Data -->
          <div>
            <data-source-meta-data
              v-if="dataSource.id"
              v-bind:id="dataSource.id"
              v-bind:createdDate="dataSource.createdDate"
              v-bind:createdBy="dataSource.userByCreatedById.email"
              v-bind:updatedDate="dataSource.updatedDate"
              v-bind:updatedBy="dataSource.userByUpdatedById.email"
            ></data-source-meta-data>
          </div>

          <!-- Button Menu -->
          <div class="mt-3">
            <data-source-button-save v-bind:dataSource="dataSource" v-bind:showPasswordField="showPasswordField"> </data-source-button-save>
            <data-source-button-reset-password v-on:resetPassword="resetPassword" v-bind:dataSourceId="dataSourceId"> </data-source-button-reset-password>
            <data-source-button-test-connectivity v-on:connectivityTestStatus="connectivityTestStatus" v-bind:dataSourceId="dataSourceId"> </data-source-button-test-connectivity>
            <data-source-button-close> </data-source-button-close>
            <data-source-button-delete v-if="dataSource.id" v-bind:dataSourceId="dataSource.id"> </data-source-button-delete>
          </div>
        </div>
      </div>
    </form>
  </div>
</template>

<script>
import DataSourceSelectDataSourceType from "./DataSourceSelectDataSourceType.vue";
import DataSourceButtonSave from "./DataSourceButtonSave.vue";
import DataSourceButtonResetPassword from "./DataSourceButtonResetPassword.vue";
import DataSourceButtonTestConnectivity from "./DataSourceButtonTestConnectivity.vue";
import DataSourceButtonClose from "./DataSourceButtonClose.vue";
import DataSourceButtonDelete from "./DataSourceButtonDelete.vue";
import MetaDataCard from "../utils/MetaDataCard.vue";
import Mixins from "../utils/Mixins.vue";

export default {
  mixins: [Mixins],
  components: {
    "data-source-select-data-source-type": DataSourceSelectDataSourceType,
    "data-source-meta-data": MetaDataCard,
    "data-source-button-save": DataSourceButtonSave,
    "data-source-button-reset-password": DataSourceButtonResetPassword,
    "data-source-button-test-connectivity": DataSourceButtonTestConnectivity,
    "data-source-button-close": DataSourceButtonClose,
    "data-source-button-delete": DataSourceButtonDelete
  },
  data: function() {
    return {
      dataSource: {},
      showPasswordField: false,
      spinner: false
    };
  },
  computed: {
    dataSourceId() {
      return this.$route.params.dataSourceId.toString();
    },
    connectivityStatusOK() {
      return this.dataSource.connectivityStatus == "Success";
    },
    connectivityStatusKO() {
      return this.dataSource.connectivityStatus == "Failed";
    }
  },
  created: function() {
    // If dataSourceId != new then get data for existing data source
    if (this.dataSourceId != "new") {
      let payload = {
        query: this.$store.state.queryGetDataSource,
        variables: {
          id: parseInt(this.dataSourceId)
        }
      };
      let headers = {};
      if (this.$session.exists()) {
        headers = { Authorization: "Bearer " + this.$session.get("jwt") };
      }
      this.$http.post(this.$store.state.graphqlUrl, payload, { headers }).then(
        function(response) {
          if (response.data.errors) {
            this.displayError(response);
          } else {
            this.dataSource = response.data.data.dataSourceById;
          }
        },
        // Error callback
        function(response) {
          this.displayError(response);
        }
      );
    } else {
      this.showPasswordField = true;
    }
  },
  methods: {
    getDataSourceType(value) {
      // Get user group from child component
      if (value != null) {
        this.dataSource["dataSourceTypeId"] = value;
      } else {
        this.list["dataSourceTypeId"] = null;
      }
    },
    resetPassword(value) {
      this.showPasswordField = value;
    },
    connectivityTestStatus(value) {
      this.spinner = value.spinner;
      if (!value.spinner) {
        this.dataSource.connectivityStatus = value.connectivityStatus;
        this.dataSource.updatedDate = value.updatedDate;
        this.dataSource.userByUpdatedById.email = value.userByUpdatedById.email;
      }
    }
  }
};
</script>
