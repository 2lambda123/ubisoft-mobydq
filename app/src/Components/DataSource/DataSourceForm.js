import React from 'react';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import TextInput from './../FormInput/TextInput';
import SelectInput from './../FormInput/SelectInput';
import SaveButton from './../FormInput/SaveButton';
import ExecuteButton from './../FormInput/ExecuteButton';
import DeleteButton from './../FormInput/DeleteButton';

const DataSourceFormFields = (props) => {
  const {
    data,
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting
  } = props;
  return <form onSubmit={handleSubmit} style={{ 'marginLeft': '50px' }}>
    <div style={{ 'marginTop': '10px', 'marginBottom': '30px' }}>
      <SaveButton disabled={isSubmitting} />
      <ExecuteButton disabled={isSubmitting} />
      <DeleteButton disabled={isSubmitting} />
    </div>
    <div>
      <TextInput
        id="name"
        label="Name"
        helperText=""
        placeholder=""
        touched={touched.name}
        error={touched.name && errors.name}
        value={values.name}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <SelectInput
        id="dataSourceTypeId"
        label="Type"
        items={data.allDataSourceTypes.nodes}
        touched={touched.dataSourceTypeId}
        error={touched.dataSourceTypeId && errors.dataSourceTypeId}
        value={values.dataSourceTypeId}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="connectionString"
        label="Connection string"
        helperText=""
        placeholder="Connection string"
        touched={touched.connectionString}
        error={touched.connectionString && errors.connectionString}
        value={values.connectionString}
        onChange={handleChange}
        onBlur={handleBlur}
        multiline={true}
        rows={5}
      />
    </div>
    <div>
      <TextInput
        id="login"
        label="Login"
        helperText=""
        placeholder=""
        touched={touched.login}
        error={touched.login && errors.login}
        value={values.login}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="password"
        label="Password"
        helperText=""
        placeholder=""
        touched={touched.password}
        error={touched.password && errors.password}
        value={values.password}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="connectivityStatus"
        label="Connectivity status"
        helperText=""
        placeholder=""
        touched={touched.connectivityStatus}
        error={touched.connectivityStatus && errors.connectivityStatus}
        value={values.connectivityStatus}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="createdDate"
        label="Created Date"
        helperText=""
        placeholder=""
        touched={touched.createdDate}
        error={touched.createdDate && errors.createdDate}
        value={values.createdDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="createdBy"
        label="Created By"
        helperText=""
        placeholder=""
        touched={touched.createdBy}
        error={touched.createdBy && errors.createdBy}
        value={values.createdBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedDate"
        label="Updated Date"
        helperText=""
        placeholder=""
        touched={touched.updatedDate}
        error={touched.updatedDate && errors.updatedDate}
        value={values.updatedDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedBy"
        label="Updated By"
        helperText=""
        placeholder=""
        touched={touched.updatedBy}
        error={touched.updatedBy && errors.updatedBy}
        value={values.updatedBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
  </form>
  ;
};

const formikEnhancer = withFormik({
  'validationSchema': Yup.object().shape({
    'name': Yup.string()
      .required('Name cannot be blank'),
    'connectionString': Yup.string()
      .required('Connection string cannot be blank'),
    'login': Yup.string()
      .required('Login cannot be blank')
  }),

  'mapPropsToValues': (props) => props.initialFieldValues === null
    ? {
        'name': '',
        'connectionString': '',
        'dataSourceTypeId': '',
        'login': '',
        'password': ''
      }
    : {
        'name': props.initialFieldValues.name,
        'connectionString': props.initialFieldValues.connectionString,
        'dataSourceTypeId': props.initialFieldValues.dataSourceTypeId,
        'login': props.initialFieldValues.login,
        'password': props.initialFieldValues.password,
        'createdDate': props.initialFieldValues.createdDate,
        'createdBy': props.initialFieldValues.userByCreatedById.email,
        'updatedDate': props.initialFieldValues.updatedDate,
        'updatedBy': props.initialFieldValues.userByUpdatedById.email
      },
  'handleSubmit': (payload, { props, setSubmitting }) => {
    setSubmitting(false);
    let variables;
    if (props.initialFieldValues === null) {
      variables = { 'dataSource': payload };
    } else {
      variables = { 'dataSourcePatch': payload, 'id': props.initialFieldValues.id };
    }
    props.mutate({
      variables
    });
  },
  'displayName': 'DataSourceForm'
});


const EnhancedDataSourceForm = formikEnhancer(DataSourceFormFields);

export default EnhancedDataSourceForm;
