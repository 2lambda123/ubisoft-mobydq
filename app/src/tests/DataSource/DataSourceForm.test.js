import React from 'react';
import { shallowWrap } from './../../setupTests';

import DataSourceForm from './../../Components/DataSource/DataSourceForm';

describe('DataSourceForm component unit test', () => {
  const touched = {
    'name': false,
    'dataSourceTypeId': false,
    'connectionString': false,
    'login': false,
    'password': false
  };
  const values = {
    'name': '',
    'dataSourceTypeId': '',
    'connectionString': '',
    'login': '',
    'password': ''
  };
  const errors = {
    'name': null,
    'dataSourceTypeId': null,
    'connectionString': null,
    'login': null,
    'password': null
  };
  const data = {
    'allDataSourceTypes': {
      'nodes': [
        {
          'id': 1,
          'value': 'mock data source'
        }
      ]
    }
  };

  const wrapper = shallowWrap(<DataSourceForm data={data} initialFieldValues={null} touched={touched} values={values} errors={errors} handleBlur={jest.fn()} handleChange={jest.fn()} handleSubmit={jest.fn()} isSubmitting={false}/>);
  it('matches snapshot', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
