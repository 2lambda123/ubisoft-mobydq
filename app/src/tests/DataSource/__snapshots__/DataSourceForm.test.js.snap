// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceForm component unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Formik
    data={
      Object {
        "allDataSourceTypes": Object {
          "nodes": Array [
            Object {
              "id": 1,
              "value": "mock data source",
            },
          ],
        },
      }
    }
    displayName="DataSourceForm"
    enableReinitialize={false}
    errors={
      Object {
        "connectionString": null,
        "dataSourceTypeId": null,
        "login": null,
        "name": null,
        "password": null,
      }
    }
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleSubmit={[Function]}
    initialFieldValues={null}
    initialValues={
      Object {
        "connectionString": "",
        "dataSourceTypeId": "",
        "login": "",
        "name": "",
        "password": "",
      }
    }
    isInitialValid={false}
    isSubmitting={false}
    onSubmit={[Function]}
    render={[Function]}
    touched={
      Object {
        "connectionString": false,
        "dataSourceTypeId": false,
        "login": false,
        "name": false,
        "password": false,
      }
    }
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
    values={
      Object {
        "connectionString": "",
        "dataSourceTypeId": "",
        "login": "",
        "name": "",
        "password": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <DataSourceFormFields
        data={
          Object {
            "allDataSourceTypes": Object {
              "nodes": Array [
                Object {
                  "id": 1,
                  "value": "mock data source",
                },
              ],
            },
          }
        }
        dirty={false}
        errors={Object {}}
        handleBlur={[Function]}
        handleChange={[Function]}
        handleReset={[Function]}
        handleSubmit={[Function]}
        initialFieldValues={null}
        initialValues={
          Object {
            "connectionString": "",
            "dataSourceTypeId": "",
            "login": "",
            "name": "",
            "password": "",
          }
        }
        isSubmitting={false}
        isValid={false}
        isValidating={false}
        registerField={[Function]}
        resetForm={[Function]}
        setError={[Function]}
        setErrors={[Function]}
        setFieldError={[Function]}
        setFieldTouched={[Function]}
        setFieldValue={[Function]}
        setFormikState={[Function]}
        setStatus={[Function]}
        setSubmitting={[Function]}
        setTouched={[Function]}
        setValues={[Function]}
        submitCount={0}
        submitForm={[Function]}
        touched={Object {}}
        unregisterField={[Function]}
        validateField={[Function]}
        validateForm={[Function]}
        validateOnBlur={true}
        validateOnChange={true}
        values={
          Object {
            "connectionString": "",
            "dataSourceTypeId": "",
            "login": "",
            "name": "",
            "password": "",
          }
        }
      />,
      "value": Object {
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "connectionString": "",
          "dataSourceTypeId": "",
          "login": "",
          "name": "",
          "password": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validate": undefined,
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
        "values": Object {
          "connectionString": "",
          "dataSourceTypeId": "",
          "login": "",
          "name": "",
          "password": "",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "allDataSourceTypes": Object {
            "nodes": Array [
              Object {
                "id": 1,
                "value": "mock data source",
              },
            ],
          },
        },
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialFieldValues": null,
        "initialValues": Object {
          "connectionString": "",
          "dataSourceTypeId": "",
          "login": "",
          "name": "",
          "password": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "values": Object {
          "connectionString": "",
          "dataSourceTypeId": "",
          "login": "",
          "name": "",
          "password": "",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {},
        "_currentValue2": Object {},
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <DataSourceFormFields
          data={
            Object {
              "allDataSourceTypes": Object {
                "nodes": Array [
                  Object {
                    "id": 1,
                    "value": "mock data source",
                  },
                ],
              },
            }
          }
          dirty={false}
          errors={Object {}}
          handleBlur={[Function]}
          handleChange={[Function]}
          handleReset={[Function]}
          handleSubmit={[Function]}
          initialFieldValues={null}
          initialValues={
            Object {
              "connectionString": "",
              "dataSourceTypeId": "",
              "login": "",
              "name": "",
              "password": "",
            }
          }
          isSubmitting={false}
          isValid={false}
          isValidating={false}
          registerField={[Function]}
          resetForm={[Function]}
          setError={[Function]}
          setErrors={[Function]}
          setFieldError={[Function]}
          setFieldTouched={[Function]}
          setFieldValue={[Function]}
          setFormikState={[Function]}
          setStatus={[Function]}
          setSubmitting={[Function]}
          setTouched={[Function]}
          setValues={[Function]}
          submitCount={0}
          submitForm={[Function]}
          touched={Object {}}
          unregisterField={[Function]}
          validateField={[Function]}
          validateForm={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
          values={
            Object {
              "connectionString": "",
              "dataSourceTypeId": "",
              "login": "",
              "name": "",
              "password": "",
            }
          }
        />,
        "value": Object {
          "dirty": false,
          "errors": Object {},
          "handleBlur": [Function],
          "handleChange": [Function],
          "handleReset": [Function],
          "handleSubmit": [Function],
          "initialValues": Object {
            "connectionString": "",
            "dataSourceTypeId": "",
            "login": "",
            "name": "",
            "password": "",
          },
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "registerField": [Function],
          "resetForm": [Function],
          "setError": [Function],
          "setErrors": [Function],
          "setFieldError": [Function],
          "setFieldTouched": [Function],
          "setFieldValue": [Function],
          "setFormikState": [Function],
          "setStatus": [Function],
          "setSubmitting": [Function],
          "setTouched": [Function],
          "setValues": [Function],
          "submitCount": 0,
          "submitForm": [Function],
          "touched": Object {},
          "unregisterField": [Function],
          "validate": undefined,
          "validateField": [Function],
          "validateForm": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "validationSchema": [Function],
          "values": Object {
            "connectionString": "",
            "dataSourceTypeId": "",
            "login": "",
            "name": "",
            "password": "",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "allDataSourceTypes": Object {
              "nodes": Array [
                Object {
                  "id": 1,
                  "value": "mock data source",
                },
              ],
            },
          },
          "dirty": false,
          "errors": Object {},
          "handleBlur": [Function],
          "handleChange": [Function],
          "handleReset": [Function],
          "handleSubmit": [Function],
          "initialFieldValues": null,
          "initialValues": Object {
            "connectionString": "",
            "dataSourceTypeId": "",
            "login": "",
            "name": "",
            "password": "",
          },
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "registerField": [Function],
          "resetForm": [Function],
          "setError": [Function],
          "setErrors": [Function],
          "setFieldError": [Function],
          "setFieldTouched": [Function],
          "setFieldValue": [Function],
          "setFormikState": [Function],
          "setStatus": [Function],
          "setSubmitting": [Function],
          "setTouched": [Function],
          "setValues": [Function],
          "submitCount": 0,
          "submitForm": [Function],
          "touched": Object {},
          "unregisterField": [Function],
          "validateField": [Function],
          "validateForm": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "values": Object {
            "connectionString": "",
            "dataSourceTypeId": "",
            "login": "",
            "name": "",
            "password": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {},
          "_currentValue2": Object {},
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
}
`;
